acceptance-tests/tests/common.go

```diff
@@ -0,0 +1,379 @@
+	e := err.(*exec.ExitError)
+	acclog.InfoLogger.Println(e.ExitCode())
+	Expect(e.ExitCode()).To(Equal(2))
+
+}
+
+// ValidateAnalseJSONNoVulns runs analyse command with json no vulns
+func ValidateAnalseJSONNoVulns() {
+	cmd := exec.Command(getCRDAcmd(), "analyse", "data"+target, "-j")
+	stdout, err := cmd.Output()
+	acclog.InfoLogger.Println(string(stdout))
+	e := err.(*exec.ExitError)
+	acclog.InfoLogger.Println(e.ExitCode())
+	Expect(e.ExitCode()).To(Equal(2))
+
+}
+
+// ValidateAnalseVulnVerbose runs analyse command with verbose
+func ValidateAnalseVulnVerbose() {
+	cmd := exec.Command(getCRDAcmd(), "analyse", "data"+target, "-v")
+	stdout, err := cmd.Output()
+	acclog.InfoLogger.Println(string(stdout))
+	Expect(err).NotTo(HaveOccurred())
+
+}
+
+// ValidateInvalidFilePath runs analyse command with invalid file path
+func ValidateInvalidFilePath() {
+	cmd := exec.Command(getCRDAcmd(), "analyse", "/package.json", "-v")
+	stdout, err := cmd.Output()
+	acclog.InfoLogger.Println(string(stdout))
+	Expect(err).To(HaveOccurred())
+}
+
+// ValidateInvalidCommand runs invalid command
+func ValidateInvalidCommand() {
+	cmd := exec.Command(getCRDAcmd(), "analysess")
+	stdout, err := cmd.Output()
+	acclog.InfoLogger.Println(string(stdout))
+	Expect(err).NotTo(HaveOccurred())
+}
+
+// ValidateInvalidFlag runs analyse command with invalid flag
+func ValidateInvalidFlag() {
+	cmd := exec.Command(getCRDAcmd(), "analyse", "-ghghd")
+	stdout, err := cmd.Output()
+	acclog.InfoLogger.Println(string(stdout))
+	Expect(err).To(HaveOccurred())
+	e := err.(*exec.ExitError)
+	Expect(e.ExitCode()).To(Equal(1))
+}
+
+// ValidateAnalseVulnDebug runs analyse command with debug
+func ValidateAnalseVulnDebug() {
+	cmd := exec.Command(getCRDAcmd(), "analyse", "data"+target, "-d")
+	cmdReader, _ := cmd.StdoutPipe()
+	scanner := bufio.NewScanner(cmdReader)
+	done := make(chan bool)
+	go func() {
+		for scanner.Scan() {
+			acclog.InfoLogger.Printf(scanner.Text())
+		}
+		done <- true
+	}()
+	cmd.Start()
+	<-done
+	err = cmd.Wait()
+	Expect(err).NotTo(HaveOccurred())
+}
+
+// ValidateAnalseAllFlags runs analyse command with all flags set true
+func ValidateAnalseAllFlags() {
+	cmd := exec.Command(getCRDAcmd(), "analyse", "data"+target, "-v", "-j", "-d")
+	stdout, err := cmd.Output()
+	acclog.InfoLogger.Println(string(stdout))
+	e := err.(*exec.ExitError)
+	acclog.InfoLogger.Println(e.ExitCode())
+	Expect(e.ExitCode()).To(Equal(2))
+
+}
+
+// CleanupNpm cleans up npm releated data
+func CleanupNpm() {
+	err := helper.CleanupNpm("data")
+	if err != nil {
+		acclog.ErrorLogger.Println(err)
+	}
+}
+
+// CleanupMvn cleans up mvn releated data
+func CleanupMvn() {
+	err := helper.CleanupMvn("data")
+	if err != nil {
+		acclog.ErrorLogger.Println(err)
+	}
+}
+
+// CleanupGo cleans up go releated data
+func CleanupGo() {
+	err := helper.CleanupGo("data")
+	if err != nil {
+		acclog.ErrorLogger.Println(err)
+	}
+}
+
+// CleanupPypi cleans up pypi releated data
+func CleanupPypi() {
```

Comment:
Ideally command `crda analyse` itself should do the cleanup.. Why we have to write a separate logic for cleanup again...?

Reply:
tests work this way -> first it picks up a manifest  copies it to the data dir-> then installs the deps -> then run analyse command -> cleanup 
cleanup refers to deletion of the copied file in the data dir

Reply:
if this has to have for every scenario..can't it be Globally..?

Reply:
yup it can be written let me change